src/main/java/it/gov/pagopa/payment/service/payment/qrcode/QRCodePreAuthServiceImpl.java

```diff
@@ -12,27 +12,23 @@
 import it.gov.pagopa.payment.utils.RewardConstants;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 
 @Service
 @Slf4j
 public class QRCodePreAuthServiceImpl extends CommonPreAuthServiceImpl implements QRCodePreAuthService {
-  private final TransactionInProgressRepository transactionInProgressRepository;
   public QRCodePreAuthServiceImpl(@Value("${app.qrCode.expirations.authorizationMinutes:15}") long authorizationExpirationMinutes,
                                   TransactionInProgressRepository transactionInProgressRepository,
                                   RewardCalculatorConnector rewardCalculatorConnector,
                                   AuditUtilities auditUtilities,
                                   WalletConnector walletConnector) {
     super(authorizationExpirationMinutes, transactionInProgressRepository, rewardCalculatorConnector, auditUtilities, walletConnector);
-    this.transactionInProgressRepository = transactionInProgressRepository;
   }
 
   @Override
   public AuthPaymentDTO relateUser(String trxCode, String userId) {
     TransactionInProgress trx = transactionInProgressRepository.findByTrxCode(trxCode.toLowerCase())
-            .orElseThrow(() -> new ClientExceptionWithBody(
-                    HttpStatus.NOT_FOUND,
+            .orElseThrow(() -> new TransactionNotFoundOrExpiredException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor